addons/mail/static/src/components/activity/tests/activity_tests.js

```diff
@@ -1276,6 +1276,52 @@ QUnit.test('data-oe-id & data-oe-model link redirection on click', async functio
     );
 });
 
+QUnit.test('chatter is updated when activity type is changed', async function (assert) {
+    assert.expect(2);
+
+    const activityId = 513;
+    this.data['res.partner'].records.push({
+        activity_ids: [activityId],
+        id: 100,
+    });
+    this.data['mail.activity'].records.push({
+        activity_category: 'upload_file',
+        activity_type_id: 28,
+        can_write: true,
+        id: activityId,
+        res_id: 100,
+        res_model: 'res.partner',
+    });
+    const { createChatterContainerComponent } = await this.start();
+    await createChatterContainerComponent({
+        threadId: 100,
+        threadModel: 'res.partner',
+    });
+
+    await this.env.services.rpc({
+        model: 'mail.activity',
+        method: 'write',
+        args: [[activityId], {
+            activity_category: 'default',
+            activity_type_id: 1,
+        }],
+    });
+    await afterNextRender(async () => {
+        const activity = this.messaging.models['Activity'].findFromIdentifyingData({ id: activityId });
+        await activity.fetchAndUpdate();
+    });
+    assert.containsOnce(
+        document.body,
+        '.o_Activity_markDoneButton',
+        "should have a mark done button when changing activity type from 'upload_file' to another type"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_FileUploader',
+        "should not have a file uploader after changing the activity type"
+    );
```

Comment:
This is more a test that "email" type has no file uploading. Please note that the assertion is no fileuploader on whole chatter, so it can become false very easily (e.g. allow file uploading in thread, or attachment box, or...).